File: src/main/java/kajipoi/entity/Work.java
package kajipoi.entity;

import kajipoi.domain.Id;
import kajipoi.domain.Status;
import lombok.Value;
import lombok.experimental.Accessors;
import org.seasar.doma.Entity;
import org.seasar.doma.jdbc.entity.NamingType;

import java.time.LocalDate;

/**
 * 仕事イベントを表すエンティティ。
 *
 * @author syobochim
 * @since 1.0
 */
@Entity(naming = NamingType.LOWER_CASE, immutable = true)
@Value(staticConstructor = "of")
@Accessors(fluent = true)
public class Work {
    private final Id memberId;
    private final Id houseworkId;
    private final LocalDate workData;
    private final Status status;
}

File: src/main/kotlin/kajipoi/entity/Work.kt
package kajipoi.entity

import kajipoi.domain.Id
import kajipoi.domain.Status
import kajipoi.domain.WorkDate
import org.seasar.doma.Entity
import org.seasar.doma.jdbc.entity.NamingType

/**
 * メンバーがやったお仕事の記録用テーブル
 * @author syobochim
 */
@Entity(naming = NamingType.LOWER_CASE, immutable = true)
class Work(val memberId: Id,
           val houseworkId: Id,
           val status: Status,
           val workDate: WorkDate)